src/persist/src/indexed/mod.rs

```diff
@@ -32,17 +32,61 @@ use timely::progress::Timestamp as TimelyTimestamp;
 
 use crate::error::Error;
 use crate::indexed::arrangement::{Arrangement, ArrangementSnapshot};
-use crate::indexed::background::Maintainer;
+use crate::indexed::background::{CompactTraceReq, CompactTraceRes, Maintainer};
 use crate::indexed::cache::BlobCache;
 use crate::indexed::columnar::ColumnarRecords;
 use crate::indexed::encoding::{
     ArrangementMeta, BlobMeta, BlobUnsealedBatch, Id, StreamRegistration, TraceBatchMeta,
     UnsealedBatchMeta,
 };
 use crate::indexed::metrics::Metrics;
-use crate::pfuture::PFutureHandle;
+use crate::pfuture::{PFuture, PFutureHandle};
 use crate::storage::{Blob, Log, SeqNo};
 
+/// A request for some work e.g. trace compaction, that can be performed outside
+/// of the main [Indexed] loop.
+#[derive(Debug)]
+pub enum ExternalWorkReq {
+    /// A request to compact a trace by merging together some immutable batches.
+    CompactTrace(CompactTraceReq),
+}
+
+/// A future for some work e.g. trace compaction, that can be performed outside
+/// of the main [Indexed] loop..
+#[derive(Debug)]
+pub enum ExternalWorkFuture {
```

Comment:
I added a commit that makes this just a Req/Res pair, but now because we don't have an intermediate type for the future I _think_ we have to clone the Maintainer and do the req -> res conversion all in the tokio task that we spawn. It's very likely that there could be a better way there - wdyt?